docs/docs/design/availability-distribution.md

```diff
@@ -0,0 +1,174 @@
+# Availability Distribution Subsystem
+
+[The availability distribution subsystem](https://paritytech.github.io/polkadot-sdk/book/node/availability/availability-distribution.html)
+is responsible for retrieving and distributing availability data such as PoVs and erasure chunks.
+
+## Subsystem Structure
+
+The implementation must conform to the `Subsystem` interface defined in the `parachaintypes` package. It should live in
+a package named `availabilitydistribution` under `dot/parachain/availability-distribution`.
+
+### Messages Received
+
+The subsystem must be registered with the overseer and handle three subsystem-specific messages from it:
+
+1. [`parachain.ChunkFetchingRequest`](https://github.com/ChainSafe/gossamer/blob/32256782470db15efb3b7b4ba687311dd4e7cdce/dot/parachain/chunk_fetching.go#L14)
+
+This is a network message received from other validators in the core group. Although request/response protocols are not
+yet supported by our network bridge implementation, the message type already exists. This should be enough to implement
+the subsystem including tests.
+
+Protocol versions `v1` and `v2` exist for chunk fetching. The `ChunkFetchingRequest` message is identical in both
+versions. The `ChunkFetchingResponse` message in `v1` omits the chunks index. In `v2` it contains [the full
+`ErasureChunk` object](https://github.com/paritytech/polkadot-sdk/blob/4c618a83d33281fe96f0e2b68a111ed227af22c0/polkadot/node/network/protocol/src/request_response/v2.rs#L99).
+[Our implementation](https://github.com/ChainSafe/gossamer/blob/32256782470db15efb3b7b4ba687311dd4e7cdce/dot/parachain/chunk_fetching.go#L89)
+only covers `v1` so far. We should probably add the chunk index to this struct and only support `v2` for the initial
+implementation, analogous the approach in other subsystems.
+
+2. [`parachain.PoVFetchingRequest`](https://github.com/ChainSafe/gossamer/blob/32256782470db15efb3b7b4ba687311dd4e7cdce/dot/parachain/pov_fetching.go#L14)
+
+This is a network message received from other validators in the core group. Although request/response protocols are not
```

Comment:
I do not think we should mention here that request response is not implemented, since it wil lbe implemented by the time somebody will start implementing it.